src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/NoResourcePublicConstructorQuickFix.java

```diff
@@ -18,64 +18,98 @@
+            codeActions.add(JDTUtils.createCodeAction(context, diagnostic, Messages.getMessage("NoargPublicConstructor"), getParticipantId()));
+
+            return codeActions;
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+
+        final CodeAction toResolve = context.getUnresolved();
+        PsiElement node = context.getCoveredNode();
+        PsiMethod parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+
+        if (parentMethod != null) {
+
             JavaCodeActionContext targetContext = context.copy();
             node = targetContext.getCoveredNode();
             PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
             parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
 
-            ChangeCorrectionProposal proposal = new ModifyModifiersProposal(TITLE_MESSAGE, targetContext.getSource().getCompilationUnit(),
-                    targetContext.getASTRoot(), parentType, 0, parentMethod.getModifierList(), Collections.singletonList("public"));
+            String message = toResolve.getTitle();
+
+            if (message.equals(Messages.getMessage("MakeConstructorPublic"))) {
+
+                ChangeCorrectionProposal proposal = new ModifyModifiersProposal(Messages.getMessage("MakeConstructorPublic"), targetContext.getSource().getCompilationUnit(),
+                        targetContext.getASTRoot(), parentType, 0, parentMethod.getModifierList(), Collections.singletonList("public"));
 
-            // Convert the proposal to LSP4J CodeAction
-            CodeAction codeAction = targetContext.convertToCodeAction(proposal, diagnostic);
-            codeAction.setTitle(TITLE_MESSAGE);
-            codeActions.add(codeAction);
+                try {
+                    WorkspaceEdit we = targetContext.convertToWorkspaceEdit(proposal);
+                    toResolve.setEdit(we);
+                } catch (Exception e) {
+                    LOGGER.log(Level.WARNING, "Unable to create workspace edit for code action to make constructor public", e);
+                }
 
-            final PsiParameterList list = parentMethod.getParameterList();
-            if (list != null && list.getParametersCount() > 0) {
-                targetContext = context.copy();
-                node = targetContext.getCoveredNode();
-                parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
+            } else if (message.equals(Messages.getMessage("NoargPublicConstructor"))) {
 
-                final String name = Messages.getMessage("NoargPublicConstructor");
-                proposal = new AddConstructorProposal(name,
+                ChangeCorrectionProposal proposal = new AddConstructorProposal(Messages.getMessage("NoargPublicConstructor"),
```

Comment:
I noticed the same messages are being retrieved/computed multiple times. The value of the message was just checked above. You can pass that string directly to the constructor.